v2/tools/generator/internal/config/group_configuration.go

```diff
@@ -91,10 +108,25 @@ func (gc *GroupConfiguration) Add(version *VersionConfiguration) *GroupConfigura
 }
 
 // findVersion uses the provided TypeName to work out which nested VersionConfiguration should be used
-func (gc *GroupConfiguration) findVersion(name astmodel.TypeName) (*VersionConfiguration, bool) {
-	v := strings.ToLower(name.PackageReference.PackageName())
-	version, ok := gc.versions[v]
-	return version, ok
+func (gc *GroupConfiguration) findVersion(name astmodel.TypeName) (*VersionConfiguration, error) {
+	ref := name.PackageReference
+	if s, ok := ref.(astmodel.StoragePackageReference); ok {
+		// If we have a storage package reference, need to unwrap the actual local package reference
+		// as all our configuration is based on API versions, not storage versions
+		ref = s.Local()
+	}
+
+	v := strings.ToLower(ref.PackageName())
+	if version, ok := gc.versions[v]; ok {
+		return version, nil
+	}
+
+	return nil, errors.Errorf(
```

Comment:
This is the error I'd change to be a specific NotFoundError type.

Reply:
Done.